### PR TITLE
Add Jenkinsfile

### PR DESCRIPTION
**Pull request by alexmuller**
Jenkins 2 builds using a Jenkinsfile.

### CODE REVIEW

**Comment by alexmuller** [comment]
Immediately after merging I'll need to set the build number for the master branch to tag.

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      govuk.rubyLinter()
+    }
+
+    stage("Build") {
+      govuk.runRakeTask('db:mongoid:drop')
+      sh "bundle exec rake"
+    }
+
+    govuk.pushTag(REPOSITORY, env.BRANCH_NAME, 'release_' + env.BUILD_NUMBER)
+    govuk.deployIntegration(REPOSITORY, env.BRANCH_NAME, 'release', 'deploy')
```
Comment (alexmuller) [inline Jenkinsfile]: Sure thing, have updated!

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    stage("Build") {
+      govuk.runRakeTask('db:mongoid:drop')
+      sh "bundle exec rake"
+    }
+
+    stage("Result") {
+      govuk.pushTag(REPOSITORY, env.BRANCH_NAME, 'release_' + env.BUILD_NUMBER)
+      govuk.deployIntegration(REPOSITORY, env.BRANCH_NAME, 'release', 'deploy')
+    }
```
Comment (cbaines) [inline Jenkinsfile]: I've been suggesting to people to make the stage conditional (e.g. [1]) if all the operations in it are conditional on the branch, as this is far less surprising both in the Jenkinsfile, and in the Jenkins web ui. As result is such a generic name, that's not as applicable here, but I thought I'd bring the option up.

1: https://github.com/alphagov/gds-api-adapters/blob/master/Jenkinsfile#L121

Comment (alexmuller) [inline Jenkinsfile]: Yeah, I've been wondering about that - those tasks are conditional in the shared Groovy code, but I wonder if that's too magic. We could also duplicate that in the application Jenkinsfile if that would make people feel better.

**Pull request merged**